components/webext-storage/src/db.rs

```diff
@@ -63,6 +68,29 @@ impl StorageDb {
         }
     }
 
+    /// Returns an interrupt handle for this database connection. This object
+    /// should be handed out to consumers that want to interrupt long-running
+    /// operations. It's FFI-safe, and `Send + Sync`, since it only makes sense
+    /// to use from another thread. Calling `interrupt` on the handle sets a
+    /// flag on all currently active interrupt scopes.
+    pub fn interrupt_handle(&self) -> SqlInterruptHandle {
+        SqlInterruptHandle::new(
+            self.writer.get_interrupt_handle(),
+            self.interrupt_counter.clone(),
+        )
+    }
+
+    /// Creates an object that knows when it's been interrupted. This should
+    /// be created at the same time as an interrupt handle, and passed to any
+    /// methods that do database work. This is the other side of the
+    /// `SqlInterruptHandle`: when a handle is interrupted, it flags all
+    /// active interrupt scopes as interrupted, too, so that they can abort
+    /// pending work.
+    #[allow(dead_code)]
```

Comment:
This isn't used yet, but I wanted to get it in here so that we don't have to add it back when we go to use it in Mark's syncing patch.